### PR TITLE
Fixes #29244: Default to disabling SSLv3

### PR DESCRIPTION
**Pull request by ehelms**
This should be dropped when EL5 EOLs.

### CODE REVIEW

File: config/foreman.hiera/security.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 ---
+apache::mod::ssl::ssl_protocol: 'all -SSLv3' # Drop when EL5 EOLs
```
Comment (ekohl) [inline config/foreman.hiera/security.yaml:2]: Note that this won't work on EL8 which has no `-SSLv3`.

**Review by ekohl — CHANGES_REQUESTED** [review]
This should already match the default in https://github.com/puppetlabs/puppetlabs-apache (which is `[ 'all', '-SSLv2', '-SSLv3' ]` in 1.24-stable).

I believe the root cause of this is an Insights bug that it doesn't look at the mod_ssl config and only at vhost itself but I can't find the exact bugzilla. Can you please provide proof that it is actually enabling SSLv3? https://nmap.org/nsedoc/scripts/ssl-enum-ciphers.html should be the best way.

**Comment by ekohl** [comment]
So I updated my nightly box and ran the script:
```
$ nmap --script ssl-enum-ciphers centos7-foreman-nightly.wisse.example.com -p 443 -v
Starting Nmap 7.80 ( https://nmap.org ) at 2020-03-03 09:52 CET
NSE: Loaded 1 scripts for scanning.
NSE: Script Pre-scanning.
Initiating NSE at 09:52
Completed NSE at 09:52, 0.00s elapsed
Initiating Ping Scan at 09:52
Scanning centos7-foreman-nightly.wisse.example.com (192.168.122.200) [2 ports]
Completed Ping Scan at 09:52, 0.00s elapsed (1 total hosts)
Initiating Parallel DNS resolution of 1 host. at 09:52
Completed Parallel DNS resolution of 1 host. at 09:52, 0.00s elapsed
Initiating Connect Scan at 09:52
Scanning centos7-foreman-nightly.wisse.example.com (192.168.122.200) [1 port]
Discovered open port 443/tcp on 192.168.122.200
Completed Connect Scan at 09:52, 0.00s elapsed (1 total ports)
NSE: Script scanning 192.168.122.200.
Initiating NSE at 09:52
Completed NSE at 09:52, 0.64s elapsed
Nmap scan report for centos7-foreman-nightly.wisse.example.com (192.168.122.200)
Host is up (0.00017s latency).

PORT    STATE SERVICE
443/tcp open  https
| ssl-enum-ciphers: 
|   TLSv1.0: 
|     ciphers: 
|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (secp256r1) - A
|       TLS_DHE_RSA_WITH_AES_128_CBC_SHA (dh 4096) - A
|       TLS_DHE_RSA_WITH_AES_256_CBC_SHA (dh 4096) - A
|       TLS_RSA_WITH_AES_128_CBC_SHA (rsa 4096) - A
|       TLS_RSA_WITH_AES_256_CBC_SHA (rsa 4096) - A
|     compressors: 
|       NULL
|     cipher preference: server
|     warnings: 
|       Key exchange (secp256r1) of lower strength than certificate key
|   TLSv1.1: 
|     ciphers: 
|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (secp256r1) - A
|       TLS_DHE_RSA_WITH_AES_128_CBC_SHA (dh 4096) - A
|       TLS_DHE_RSA_WITH_AES_256_CBC_SHA (dh 4096) - A
|       TLS_RSA_WITH_AES_128_CBC_SHA (rsa 4096) - A
|       TLS_RSA_WITH_AES_256_CBC_SHA (rsa 4096) - A
|     compressors: 
|       NULL
|     cipher preference: server
|     warnings: 
|       Key exchange (secp256r1) of lower strength than certificate key
|   TLSv1.2: 
|     ciphers: 
|       TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (secp256r1) - A
|       TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 (dh 4096) - A
|       TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 (dh 4096) - A
|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (secp256r1) - A
|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (secp256r1) - A
|       TLS_DHE_RSA_WITH_AES_128_CBC_SHA256 (dh 4096) - A
|       TLS_DHE_RSA_WITH_AES_128_CBC_SHA (dh 4096) - A
|       TLS_DHE_RSA_WITH_AES_256_CBC_SHA256 (dh 4096) - A
|       TLS_DHE_RSA_WITH_AES_256_CBC_SHA (dh 4096) - A
|       TLS_RSA_WITH_AES_128_GCM_SHA256 (rsa 4096) - A
|       TLS_RSA_WITH_AES_256_GCM_SHA384 (rsa 4096) - A
|       TLS_RSA_WITH_AES_128_CBC_SHA256 (rsa 4096) - A
|       TLS_RSA_WITH_AES_256_CBC_SHA256 (rsa 4096) - A
|       TLS_RSA_WITH_AES_128_CBC_SHA (rsa 4096) - A
|       TLS_RSA_WITH_AES_256_CBC_SHA (rsa 4096) - A
|     compressors: 
|       NULL
|     cipher preference: server
|     warnings: 
|       Key exchange (secp256r1) of lower strength than certificate key
|_  least strength: A

NSE: Script Post-scanning.
Initiating NSE at 09:52
Completed NSE at 09:52, 0.00s elapsed
Read data files from: /usr/bin/../share/nmap
Nmap done: 1 IP address (1 host up) scanned in 0.84 seconds
```

I've tried to search for the Insights conversation, but can't find it. I believe Insights still only reads the vhosts and maybe the main httpd config. I really think this should be bounced back to the Insights rules.

**Comment by ehelms** [comment]
Thanks for digging in. That command is quite useful!

**Pull request closed without merging**